### ISSUE TITLE
Error 2106 - AuthorizationCodeInvalid

### ISSUE THREAD

**Issue by etiennemartin** [Bungie-net/api#32]
Hey everyone,
I've been beating my head against the wall with the authentication process. I'm attempting to use the code provided by the re-direct to get the api token (api, refresh token) and I keep getting the same error message from Bungie:

```
{
  ErrorCode = 2106;
  ErrorStatus = AuthorizationCodeInvalid;
  Message = "Authorization code is invalid.";
  MessageData = {};
  ThrottleSeconds = 0;
}
```

The code given to be looks like this: M11A1A11A-A1A1-11A1-11A1-A1A11AA11A1A which doesn't match the examples I see (I've swapped all letters and numbers for As and 1s). Is this a valid format or do I have to do something to it? Like hashing it or something?

Note I'm trying this using an Xbox Login.

**Comment by vthornheart-bng** [comment]
Ah, feel free to comment here if you found out what was going wrong, just in case it will help others!

**Comment by etiennemartin** [comment]
Sorry about the delay. I closed this issue when I realized I was doing something wrong. Now that I've solved it I feel like I can justify an answer in case someone else runs into it. So here goes.

Given I'm developing for iOS I'm using a web view to do authentication. My application (https://www.bungie.net/en/Application) wasn't setup properly to begin with. I had the Application Authentication setup wrong. I had it set to `OAUTH Client Type` to not applicable, and no redirect URL. On top of that I wasn't sending in the proper `client_id` as a parameter when I was opening up the authentication URL (https://www.bungie.net/en/OAuth/Authorize?client_id={client-id}&response_type=code). The odd thing is the authentication page that loaded looked fine and allowed me to pick a service to login with even if I wasn't setup right. Now that I have it setup right everything works fine and the codes look the same as the examples.

Using the Demo here: https://lowlidev.com.au/destiny/authentication-2 really helped debug the issue.

One finally note, the reason why I was seeing an odd code with the format `M11A1A11A-A1A1-11A1-11A1-A1A11AA11A1A` seemed to be caused by the fact that I was going through the Microsoft OAuth flow and the code being returned was the token for Xbox Live and not Bungie's API. From what I can tell that's what I was seeing. So if you see a code that starts with "M" and has dashes it might be it.

I really hope this helps. Again sorry for closing this issue with no response. I just wanted to make sure I understood more before going on about my issue.
